src/components/transactions/Faucet/CaptchaFaucetModalContent.tsx

```diff
@@ -0,0 +1,130 @@
+  const [captchaLoading, setCaptchaLoading] = useState<boolean>(true);
+  const [txHash, setTxHash] = useState<string>('');
+  const [error, setError] = useState<string>('');
+
+  // TODO: just for testing
+  // const faucetUrl = `${process.env.NEXT_PUBLIC_API_BASEURL}/faucet`;
+  const faucetUrl = 'https://api-v2-feat-faucet-relay.aave.com/faucet';
+  const siteKey = process.env.NEXT_PUBLIC_TURNSTILE_SITE_KEY as string;
+
+  const poolReserve = reserves.find(
+    (reserve) => reserve.underlyingAsset === underlyingAsset
+  ) as ComputedReserveData;
+
+  const normalizedAmount = getNormalizedMintAmount(poolReserve.symbol, poolReserve.decimals);
+
+  const captchaVerify = (token: string) => {
+    setCaptchaToken(token);
+    setCaptchaLoading(false);
+  };
+
+  const faucet = async () => {
+    try {
+      setTxHash('');
+      setLoading(true);
+      setError('');
+      const response = await fetch(faucetUrl, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          address: account,
+          captchaToken,
+          market: currentMarket,
+          tokenAddress: poolReserve.underlyingAsset,
+          tokenSymbol: poolReserve.symbol,
+          faucetAddress: currentMarketData.addresses.FAUCET,
+        }),
+      });
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const data = await response.json();
+      setTxHash(data.msg);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        setError(e.message);
+      } else {
+        setError('An error occurred trying to send the transaction');
+      }
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  if (txHash) {
+    return (
+      <TxSuccessView
+        txHash={txHash}
+        action={<Trans>will receive</Trans>}
+        symbol={poolReserve.symbol}
+        amount={normalizedAmount}
+      />
+    );
+  }
+
+  return (
+    <>
+      <Turnstile sitekey={siteKey} onVerify={captchaVerify} autoResetOnExpire />
+      <Typography variant="h2" sx={{ mb: 6 }}>
+        <Trans>Faucet</Trans> {poolReserve.symbol}
+      </Typography>
+      <Box
+        sx={(theme) => ({
+          p: 3,
+          border: `1px solid ${theme.palette.divider}`,
+          borderRadius: '4px',
+          '.MuiBox-root:last-of-type': {
+            mb: 0,
+          },
+        })}
+      >
+        <DetailsNumberLine
+          description={<Trans>Amount</Trans>}
+          iconSymbol={poolReserve.symbol}
+          symbol={poolReserve.symbol}
+          value={normalizedAmount}
+        />
+      </Box>
+      <Typography variant="helperText" color="error.main">
+        {error}
+      </Typography>
+      <Box sx={{ display: 'flex', flexDirection: 'column', mt: 12 }}>
+        <Button
+          variant="contained"
+          disabled={loading || !captchaToken || readOnlyModeAddress !== undefined}
```

Comment:
I think complementary to this design is that we should have a message saying why it's disabled, probably as yellow warning helper text, small, underneath button.

Reply:
It's mostly useful for showing because, as a user, my UX/UI looks/feels the same between v2 and v3 markets, and it would be confusing to see it disabled between the same network, also between dev/prod environments even or other forks of our UI that may launch on other domains.  So something like:

```
if (loading) // no text needed

if (!captchToken) // "The current domain is not configured to use this faucet"

if (readOnlyModAddress) // "Faucets are not supported in read-only mode"
```

Could help with community debugging I can see too, i.e. "WHy can't I pull faucet money in my PR? It shows this message"

"What' is your URL?"

"https://my-fun-app.aave-fork.dev"

"The v3 faucets are configured for our production UI domain only"

etc.


cc: @foodaka